src/core/BUILD

```diff
@@ -1302,14 +1349,39 @@ grpc_cc_library(
     ],
 )
 
+grpc_cc_library(
+    name = "event_engine_interface",
```

Comment:
If we want to keep one target per header, this should be a public target in the root `BUILD` file. Any target that exposes `include/grpc/event_engine/*` should be exposed publicly, so third-party EventEngine implementations can depend on the right targets in their bazel builds.

Reply:
agree this should move - i'd argue that since this sprung out of an existing library in src/core let's keep it there for this pr and move it in the next one.